app.js

```diff
@@ -0,0 +1,113 @@
+/**
+ * Created by dusting on 10/26/15.
+ */
+var getFile = require('./getFile');
+
+var investigation = getFile.processArgs(process);
+
+var graphDict = {};
+var graphDictArray = [];
+var touchedDict = {seen: [], starts:[], ends:[]};
+var pathDict = {paths:[]};
+
+witnessAccounts(investigation, findPaths);
+
+
+//This section maps nodes and edges to an object
+function witnessAccounts(investigation, callback) {
+    for (var people = 0; people < investigation.length; people++) {
+        mapNodes(investigation[people]);
+    }
+    callback();
+}
+
+function mapNodes(events){
+
+    for (var i = 0; i<events.length; i++){
+
+        var node = events[i];
+        var edge = events[i+1];
+        if (graphDict.hasOwnProperty(node)){
+            addEdges(node, edge, i, events.length);
+        }else{
+            addNode(node);
+            addEdges(node, edge, i, events.length);
+        }
+    }
+}
+
+function addNode(v){
+    graphDict[v] = []; //Initialize a vertex with empty edges
+    graphDictArray.push(v);
+}
+
+function addEdges(node, edge, count, uBound){
+    if (edge != undefined && count <= uBound){
+        graphDict[node].push(edge);
+    }
+}
+
+
+//This section follows the edges from the root and creates the path arrays
+function findPaths() {
```

Comment:
Another hint for you. There may be multiple longest paths.  Each longest path represents a valid output.